src/hooks/useColorForAsset.ts

```diff
@@ -21,12 +22,15 @@ export default function useColorForAsset(
   const token = getTokenMetadata(mainnet_address || address!);
   const tokenListColor = token?.color;
 
-  const { color: imageColor } = useImageMetadata(
-    getUrlForTrustIconFallback(
-      mainnet_address || address!,
-      mainnet_address ? AssetType.token : (type as AssetType)
-    )
-  ) as any;
+  const { result: imageColor } = usePersistentDominantColorFromImage(
+    maybeSignUri(
+      getUrlForTrustIconFallback(
+        mainnet_address || address!,
+        mainnet_address ? AssetType.token : (type as AssetType)
+      ) || '',
+      { w: 40 }
+    ) || ''
+  );
```

Comment:
@skylarbarrera Would love fro you to try checking CPU usage after and before the change using XCode Instruments, I could record a loom how to use it if you wish